lib/upgradable-multisig/multisig/validation.ak

```diff
@@ -0,0 +1,173 @@
+use aiken/list
+use aiken/transaction.{Output, OutputReference, ScriptContext, Transaction}
+use aiken/transaction/credential
+use aiken/transaction/value.{PolicyId, from_minted_value}
+use aiken_design_patterns/singular_utxo_indexer as singular_utxo_indexer
+use assist/data
+use assist/values
+use upgradable_multisig/multisig/utils.{
+  get_asset_amount, signed_within_threshold, validate_multisig_datum,
+  validate_nft_output,
+}
+use upgradable_multisig/types.{MultisigDatum}
+
+pub fn validate_initiate_multisig(
+  own_policy: PolicyId,
+  output_reference: OutputReference,
+  input_index: Int,
+  output_index: Int,
+  tx: Transaction,
+) -> Bool {
+  let minted_tokens = tx.mint |> from_minted_value()
+  expect Some(own_input) = list.at(tx.inputs, input_index)
+  expect Some(own_output) = list.at(tx.outputs, output_index)
+  let multisig_credential = credential.ScriptCredential(own_policy)
+  // Ensure this outref is spent in this transaction
+  let is_outref_spent = own_input.output_reference == output_reference
+
+  // Extract datums
+  expect multisig_datum: MultisigDatum = data.output_datum(own_output)
+
+  // Create a unique token name
+  let sig_tkn_name =
+    values.unique_token_name(
+      output_reference.transaction_id.hash,
+      output_reference.output_index,
+      #"",
+    )
+  // Construct one Multisig token value 
+  let multisig_value = value.from_asset(own_policy, sig_tkn_name, 1)
+
+  // Validate internal consistency of multisig datum
+  let valid_multisig_datum = validate_multisig_datum(multisig_datum)
+
+  // Ensure token output goes back to the script
+  let multisig_token_to_script =
+    validate_nft_output(
+      own_output,
+      own_policy,
+      sig_tkn_name,
+      multisig_credential,
+    )
```

Comment:
Should we check entire output value? (token dust attack)

Reply:
I'll check the Ada as well to prevent tocken dust attack.